src/evo/deterministicmns.h

```diff
@@ -428,14 +442,24 @@ class CDeterministicMNList
         return true;
     }
     template <typename T>
-    [[nodiscard]] bool DeleteUniqueProperty(const CDeterministicMN& dmn, const T& oldValue)
+    [[nodiscard]] bool DeleteUniqueProperty(const CDeterministicMN& dmn, const T& oldValue, std::optional<bool> specific_legacy_bls_scheme = std::nullopt)
     {
         static const T nullValue;
         if (oldValue == nullValue) {
             return false;
         }
 
-        auto oldHash = ::SerializeHash(oldValue);
+        uint256 oldHash;
+        if constexpr (std::is_same<T, CBLSLazyPublicKey>()) {
+            assert(specific_legacy_bls_scheme.has_value());
+            CBLSPublicKey local_pub_key = oldValue.Get();
+            CBLSPublicKeyVersionWrapper wrapped_pub_key(local_pub_key, specific_legacy_bls_scheme.value());
+            oldHash = ::SerializeHash(wrapped_pub_key);
+        } else {
+            assert(!specific_legacy_bls_scheme.has_value());
```

Comment:
This must be removed or changed to:
```suggestion
if (specific_legacy_bls_scheme.has_value()) assert(!specific_legacy_bls_scheme.value();
```